server/api/controllers/locker-signout-controller.ts

```diff
@@ -0,0 +1,145 @@
+import { Request, Response } from "express";
+import { Logger } from "../../util/logger";
+// import { ReadWriteAPIController } from "./read-write-api-controller";
+import { ReadWriteController } from "./read-write-controller";
+import LockerData from "../../data/_hidden/locker-list.json";
+import { Locker } from "../../types/locker";
+import fs from "fs";
+
+export class LockerSignoutController {
+    static logger = new Logger("Locker Signout");
+
+    private static updateLockerStatus(term: string, userId: number, lockerNumber: number): boolean {
+      const lockerData = this.readLockerData();
+  
+      const targetTerm = lockerData[term];
+      const targetLocker = targetTerm.find((locker: Locker) => locker.lockerNumber === lockerNumber);
+  
+      if (targetLocker && targetLocker.userId === 0) {
+        targetLocker.userId = userId;
+        this.writeLockerData(lockerData); 
+        return true;
+      }
+  
+      return false;
+    }
+  
+    private static readLockerData(): { [key: string]: Locker[] } {
+      const jsonData = fs.readFileSync("server/data/_hidden/locker-list.json", "utf-8");
+      return JSON.parse(jsonData);
+    }
+  
+    private static writeLockerData(lockerData: { [key: string]: Locker[] }): void {
+      const jsonData = JSON.stringify(lockerData, null, 2);
```

Comment:
looks like you could use `this.generateJSON` here?

alternatively, this `this.generateJSON` is currently unused, it's maybe safe to delete lol